### DISCUSSION TITLE
bug? Can't read contract data with useContractRead hook

### DISCUSSION THREAD

**Discussion by izcoser** [wevm/wagmi#1073] [Q&A]
I am trying to read a user allowance on a contract but the hook never returns, neither with error or success. The loading and error variables remain false and data is undefined.

This is what the code looks like:

```js

  // connect to network code is omitted.
  const { data: allowance, isError: allowanceError, isLoading: allowanceLoading } = useContractRead({
    address: tokens[chain?.name].address,
    abi: tokens[chain?.name].abi,
    functionName: "allowance",
    args: [address, bridges[chain?.name].address],
    onSettled(data, error) {
      console.log('Settled', { data, error })
    },

  });

  console.log({ allowance });
  console.log({ allowanceError });
  console.log({ allowanceLoading });
```

I would expect it to either throw an error with onSettled or fetch the allowance successfully.
Am I doing something clearly wrong here?
I've tried inputting the fixed contract values there as well just to make sure and it's still behaving unexpectedly.

**Comment by tmm** [comment]
If you have a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), you can [create an issue](https://github.com/wagmi-dev/wagmi/issues/new?assignees=&labels=&template=bug_report.yml&title=bug%3A+%3Ctitle%3E) and we'll look into it.

**Reply by izcoser** [reply]
I've created two issues with what I hope to be good enough reproducible examples, thank you!
